llpc/util/llpcElfWriter.cpp

```diff
@@ -735,6 +737,285 @@ template <class Elf> void ElfWriter<Elf>::updateMetaNote(Context *pContext, cons
   pNewNote->data = data;
 }
 
+// =====================================================================================================================
+// Get the count of relocs in the reloc section
+//
+// @param pRelocSection : Reloc section
+template <class Elf> size_t ElfWriter<Elf>::getRelocsCount(const SectionBuffer *pRelocSection) {
+  size_t relocCount = 0;
+  if (pRelocSection->secHead.sh_size > 0) {
+    relocCount = static_cast<unsigned>(pRelocSection->secHead.sh_size / pRelocSection->secHead.sh_entsize);
+  }
+
+  return relocCount;
+}
+
+// =====================================================================================================================
+// Get the start pos of relocs used by PS in the reloc section
+//
+// @param pRelocSection : Reloc section
+// @param psIsaOffset : Ps ISA offset in the elf bin
+template <class Elf> size_t ElfWriter<Elf>::getRelocPsStartPos(const SectionBuffer *pRelocSection, size_t psIsaOffset) {
+  size_t psStartPos = 0;
+
+  size_t relocCount = getRelocsCount(pRelocSection);
+  if (relocCount > 0) {
+    auto relocs = reinterpret_cast<typename Elf::Reloc *>(const_cast<uint8_t *>(pRelocSection->data));
+    while (psStartPos < relocCount) {
+      if (relocs[psStartPos].r_offset >= psIsaOffset)
+        break;
+      psStartPos++;
+    }
+  }
+
+  return psStartPos;
+}
+
+// =====================================================================================================================
+// Create a new empty section based on the input section
+//
+// @param [in] pName : Name of the new section
+// @param [in] pInputSection : the input section
```

Comment:
I think we implicitly assume `[in]`, don't we @amdrexu?

Reply:
Also, please capitalize the first words after `:`

Reply:
Addressed.